contentcuration/contentcuration/viewsets/websockets/consumers.py

```diff
@@ -0,0 +1,133 @@
+import json
+import logging as logger
+
+from asgiref.sync import async_to_sync
+from channels.generic.websocket import WebsocketConsumer
+
+from contentcuration.models import Change
+from contentcuration.models import Channel
+from contentcuration.tasks import get_or_create_async_task
+from contentcuration.viewsets.sync.constants import CHANNEL
+from contentcuration.viewsets.sync.constants import CREATED
+
+
+logging = logger.getLogger(__name__)
+
+
+class SyncConsumer(WebsocketConsumer):
+
+    # Checks permissions
+    def check_authentication(self):
+        return self.scope['user'].is_authenticated
+
+    # Initial reset
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.room_name = None
+        self.room_group_name = None
+        self.session_key = None
+        self.user = None
+        self.user_id = None
+        self.indiviual_room_group_name = None
+
+    def connect(self):
+        """
+        Executes when a user tries to make a websocket connection.
+        - Creates and joins a group for indiviual user
+        - Joins a public group based on channel_id provided in url
+        """
+        # Extract the channel_id from url
+        self.room_name = self.scope['url_route']['kwargs']['channel_id']
+        self.room_group_name = self.room_name
+
+        logging.debug("Connected to channel_id: " + self.room_name)
+
+        self.user = self.scope["user"]
```

Comment:
I see several properties being set from information available in `self.scope`. Would it make sense to have these be class properties? for example
```python
@property
def user(self):
    return self.scope["user"]

Reply:
If you did this, `check_authentication` could use `self.user`, instead of duplicating access to the where the user object originates from